kernel/src/table_changes/log_replay/tests.rs

```diff
@@ -0,0 +1,514 @@
+use super::TableChangesScanData;
+use crate::actions::deletion_vector::DeletionVectorDescriptor;
+use crate::actions::{Add, Cdc, Metadata, Protocol, Remove};
+use crate::engine::sync::SyncEngine;
+use crate::log_segment::LogSegment;
+use crate::path::ParsedLogPath;
+use crate::scan::state::DvInfo;
+use crate::schema::{DataType, StructField, StructType};
+use crate::table_changes::log_replay::LogReplayScanner;
+use crate::utils::test_utils::LocalMockTable;
+use crate::{DeltaResult, Engine, Error, Version};
+
+use itertools::Itertools;
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::Arc;
+
+fn get_schema() -> StructType {
+    StructType::new([
+        StructField::new("id", DataType::LONG, true),
+        StructField::new("value", DataType::STRING, true),
+    ])
+}
+
+fn get_segment(
+    engine: &dyn Engine,
+    path: &Path,
+    start_version: Version,
+    end_version: impl Into<Option<Version>>,
+) -> DeltaResult<Vec<ParsedLogPath>> {
+    let table_root = url::Url::from_directory_path(path).unwrap();
+    let log_root = table_root.join("_delta_log/")?;
+    let log_segment = LogSegment::for_table_changes(
+        engine.get_file_system_client().as_ref(),
+        log_root,
+        start_version,
+        end_version,
+    )?;
+    Ok(log_segment.ascending_commit_files)
+}
+
+fn result_to_sv(iter: impl Iterator<Item = DeltaResult<TableChangesScanData>>) -> Vec<bool> {
+    iter.map_ok(|scan_data| scan_data.selection_vector.into_iter())
+        .flatten_ok()
+        .try_collect()
+        .unwrap()
+}
+
+#[tokio::test]
+async fn metadata_protocol() {
+    let engine = Arc::new(SyncEngine::new());
+    let mut mock_table = LocalMockTable::new();
+    let schema_string = serde_json::to_string(&get_schema()).unwrap();
+    mock_table
+        .commit([
+            Metadata {
+                schema_string,
+                configuration: HashMap::from([
+                    ("delta.enableChangeDataFeed".to_string(), "true".to_string()),
+                    (
+                        "delta.enableDeletionVectors".to_string(),
+                        "true".to_string(),
+                    ),
+                    ("dela.columnMapping.mode".to_string(), "none".to_string()),
```

Comment:
```suggestion
                    ("delta.columnMapping.mode".to_string(), "none".to_string()),
```

Reply:
though it's doing the same thing either way lol

Reply:
I think this is a good motivation to have these as constants somewhere 🤔 

We reference these all over the deserialize, table properties tests, log replay tests, and some of the visitors tests.

Reply:
yea we could introduce some sort of intermediate enum for `TableProperties` perhaps

opened an issue: https://github.com/delta-io/delta-kernel-rs/issues/569